lib/salus/scanners/osv/base.rb

```diff
@@ -0,0 +1,131 @@
+require 'salus/scanners/base'
+require 'json'
+require 'zip'
+require 'uri'
+require 'net/http'
+
+module Salus::Scanners::OSV
+  class Base < Salus::Scanners::Base
+    class ApiTooManyRequestsError < StandardError; end
+
+    MAX_RETRIES = 2
+
+    def run
+      raise NoMethodError
+    end
+
+    def should_run?
+      raise NoMethodError
+    end
+
+    private
+
+    def name
+      self.class.name.sub('Salus::Scanners::OSV::', '')
+    end
+
+    def osv_vulnerabilities
+      @osv_vulnerabilities ||= fetch_vulnerabilities
+    end
+
+    def osv_url
```

Comment:
This method will only return the url for a single language.  
But a repo may have multiple languages, like repo has go.sum, requirements.txt,  and Gemfile.lock.

This method should instead return urls for all applicable languages, right?